libs/langchain/langchain/text_splitter.py

```diff
@@ -1489,3 +1490,140 @@ def __init__(self, **kwargs: Any) -> None:
         """Initialize a LatexTextSplitter."""
         separators = self.get_separators_for_language(Language.LATEX)
         super().__init__(separators=separators, **kwargs)
+
+
+def calculate_serialized_size(data: Dict) -> int:
+    """Calculate the size of the serialized JSON object."""
+    return len(json.dumps(data))
+
+
+class JsonChunks(list):
+    def to_string(self) -> List[str]:
+        """Convert chunks to a list of strings."""
+        return [json.dumps(chunk) for chunk in self]
+
+    def to_documents(self, **metadata: Any) -> List[Document]:
+        return [
+            Document(page_content=chunk_str, metadata=metadata)
+            for chunk_str in self.to_string()
+        ]
+
+    def __repr__(self) -> str:
+        return json.dumps(
+            {
+                "JSON Chunks": {
+                    "chunks": len(self),
+                    "sizes": [calculate_serialized_size(chunk) for chunk in self],
+                }
+            },
+            indent=4,
+        )
+
+
+class RecursiveJsonTextSplitter:
+    """
+    This text splitter splits a json object into smaller json objects
+
+    This is necessary when splitting a nested json object to ensure that
+     the key:value relationships are preserved
+
+    """
+
+    def __init__(
+        self, max_chunk_size: int = 2000, min_chunk_size: Optional[int] = None
+    ):
+        self.max_chunk_size = max_chunk_size
+        self.min_chunk_size = (
+            min_chunk_size
+            if min_chunk_size is not None
+            else max(max_chunk_size - 200, 50)
+        )
+        self._chunks = JsonChunks()
```

Comment:
i dont think this class should be stateful - pretty inconsistent with other classes

Reply:
I get that. I think I initially coded it up for my own use. And with the chunks passed to 3 of the functions in the class I was thinking it made more sense to just have those be a class member. I can change that.

Reply:
I'll try to have it inherit from TextSplitter, and change split to split_text so that create_documents() can work.

Reply:
The issue I am seeing is with create_documents()
![image](https://github.com/langchain-ai/langchain/assets/43421397/a72fa309-e055-43f9-9d28-1cb77c444869)
The super is expecting List[str]
I wanted to keep the loaded json as a Dict rather than expecting str because the recursion for preprocessing and splitting is on Dict objects. And anyone with json in python probably has it in a Dict format to start. I think I will stop inheriting from TextSplitter, and instead just shaddow the functionality with split_text, and create_documents, with that said I'll add split_json that returns List[dict] so that people can get the output in json if they want. Then split_text and create_docs can be wrappers around split_json.

Reply:
@hwchase17 I refactored this and updated the docs and unit tests.